common/functions/src/scalars/geos/geo_to_h3.rs

```diff
@@ -0,0 +1,160 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+use std::fmt::Formatter;
+
+use common_datavalues::prelude::*;
+use common_datavalues::ColumnBuilder;
+use common_datavalues::ColumnRef;
+use common_datavalues::ColumnsWithField;
+use common_datavalues::DataTypeImpl;
+use common_datavalues::ScalarColumn;
+use common_datavalues::Series;
+use common_datavalues::TypeID;
+use common_exception::ErrorCode;
+use geo_types::Coordinate;
+use h3ron::H3Cell;
+use h3ron::Index;
+
+use crate::scalars::Function;
+use crate::scalars::FunctionContext;
+use crate::scalars::FunctionDescription;
+use crate::scalars::FunctionFeatures;
+
+#[derive(Clone)]
+pub struct GeoToH3Function {
+    display_name: String,
+}
+
+impl GeoToH3Function {
+    pub fn try_create(
+        display_name: &str,
+        args: &[&DataTypeImpl],
+    ) -> common_exception::Result<Box<dyn Function>> {
+        let mut arg = args[0];
+        if arg.data_type_id() != TypeID::Float64 {
+            return Err(ErrorCode::IllegalDataType(format!(
+                "Invalid type {} of argument {} for function '{}'. Must be Float64",
+                arg.data_type_id(),
+                1,
+                display_name,
+            )));
+        }
+
+        arg = args[1];
+        if arg.data_type_id() != TypeID::Float64 {
+            return Err(ErrorCode::IllegalDataType(format!(
+                "Invalid type {} of argument {} for function '{}'. Must be Float64",
+                arg.data_type_id(),
+                2,
+                display_name,
+            )));
+        }
+
+        arg = args[2];
+        if arg.data_type_id() != TypeID::UInt8 {
+            return Err(ErrorCode::IllegalDataType(format!(
+                "Invalid type {} of argument {} for function '{}'. Must be Float64",
+                arg.data_type_id(),
+                1,
+                display_name,
+            )));
+        }
+
+        Ok(Box::new(Self {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic().num_arguments(3))
+    }
+}
+
+impl Function for GeoToH3Function {
+    fn name(&self) -> &str {
+        &self.display_name
+    }
+
+    fn return_type(&self) -> DataTypeImpl {
+        UInt64Type::new_impl()
+    }
+
+    fn eval(
+        &self,
+        _func_ctx: FunctionContext,
+        columns: &ColumnsWithField,
+        input_rows: usize,
+    ) -> common_exception::Result<ColumnRef> {
+        // check datatype
+        let mut dt = columns[0].field().data_type();
+        if dt.data_type_id() != TypeID::Float64 {
+            return Err(ErrorCode::BadDataValueType(format!(
```

Comment:
we  can accept `numeric datatypes` and use macro to dispatch.

see: [ trigonometric.rs](https://github.com/datafuselabs/databend/blob/89d004c0ee816fbf3209ffe72b1bfcda629b86fb/common/functions/src/scalars/maths/trigonometric.rs)